### ISSUE TITLE
Add ARIA attributes to DOI display throughout Janeway interface

### ISSUE THREAD

**Issue by alainna** [openlibhums/janeway#3259]
**Is your feature request related to a problem? Please describe.**

Crossref are proposing an update to their display guideline to add ARIA attributes to the HTML used to display DOIs. As Janeway administrators and Crossref members who support websites which are accessible to all, we would like to run a trial of their proposal. 

[Crossref rationale](https://www.crossref.org/blog/accessibility-for-crossref-doi-links-call-for-comments-on-proposed-new-guidelines/): 

> DOIs and URLs themselves don’t really tell readers much. People with visual impairments rely on screen readers to read out loud the contents of a page. We’re asking for the title of each DOI to be added, in an ARIA (Accessible Rich Internet Applications) attribute, so these users understand what these links are for.
> Accessible text, as this kind of description is known, should be included for all links, but at this time, we’re specifically recommending it for landing pages of newly registered records.

**Describe the solution you'd like**
Wherever Hyperlinked DOIs are displayed in the Janeway user interface, add ``aria-label`` with the text ``DOI for [object type]: [object title]``.

Examples: 

* Journal article: `` <a href="https://doi.org/10.5555/12345678" aria-label="DOI for journal article: Toward a Unified Theory of High-Energy Metaphysics: Silly String Theory">https://doi.org/10.5555/12345678</a>``
* Preprint: ``  <a href="https://doi.org/10.5557/12345670" aria-label="DOI for preprint: Silly string and its application to pies">https://doi.org/10.5557/12345670</a>``

**Additional context**

Crossref only recommend the format "DOI for [object title]", however we recommend adding the extra "[object type]" data based on some feedback from Google Scholar during a recent meeting.

**Comment by S-Haime** [comment]
@StephDriver Is this by chance being picked up as part of the a11y work?

**Comment by S-Haime** [comment]
Thank you, that's really helpful :) I'll send it to triage and I agree to the points made above.

(@alainna I am sure you will be reading all this, but it may also be worth checking out the PR Steph has linked above :) )

**Comment by StephDriver** [comment]
Ok, this ended up sending me down a rabbit hole - I stand by `described-by` as being better for doi a11y, but in practical terms it doesn't work for us right now, and we are doing `label` instead.  This is due to the template structure and I've made further notes on #4777 for anyone wondering about it.  While it was a rabbit hole - it was also pretty satisfying to have come out of it more confident in my conclusion having done all that work, so thanks for prodding this!

In regards to the object part, we had a short inconclusive discussion during Backlog Refinement today, and we have put it on the Triage list to be reviewed in due course.  In terms of aria-labels without the object part, but that point they should be in the code!
